### ISSUE TITLE
Eigensolvers for non-symmetric matrices

### ISSUE THREAD

**Issue by maxscheurer** [adc-connect/adcc#135]
According to @mfherbst, the current eigensolver implementations (Davidson, Lanczos) assume symmetric matrices, but they can be adapted to solve eigenvalue problems of non-symmetric matrices (required for EOM etc.).

**Comment by mfherbst** [comment]
How close to needing this are you @maxscheurer?

**Comment by maxscheurer** [comment]
I don't really _need_ it for anything urgent, moving the gradients forward seems more important right now. (EOM)-CC is more of a fun playground project 😄

**Comment by mfherbst** [comment]
Agree. Thanks, than I'll review that next (sorry about the delays).

**Comment by maxscheurer** [comment]
> (sorry about the delays).

No worries at all 👍

**Comment by obackhouse** [comment]
> I don't really _need_ it for anything urgent, moving the gradients forward seems more important right now. (EOM)-CC is more of a fun playground project smile

I actually implemented a pretty simply CCSD solver in my `adcc` fork [here](https://github.com/obackhouse/adcc/tree/ccsd/adcc/cc), I wanted to implemented EOM-CCSD (for fun) but only realised `adcc` didn't have non-Hermitian solvers after I'd implemented the ground state :sweat_smile:

**Comment by maxscheurer** [comment]
Nice! I have autogenerated code for CC2-LR and EOM-CC, too... also just for fun 😬😂

**Comment by mfherbst** [comment]
sounds like we should throw stuff together and get CC working in adcc ;)

**Comment by mfherbst** [comment]
I mean the first attempt does not need to be "good".
